pkg/util/pki/parse.go

```diff
@@ -360,3 +363,52 @@ func (c *chainNode) root() *chainNode {
 func isSelfSignedCertificate(cert *x509.Certificate) bool {
 	return cert.CheckSignatureFrom(cert) == nil
 }
+
+// Copied from pkix.attributeTypeNames and inverted. (Sadly it is private.)
+var attributeTypeNames = map[string][]int{
+	"C":            {2, 5, 4, 6},
+	"O":            {2, 5, 4, 10},
+	"OU":           {2, 5, 4, 11},
+	"CN":           {2, 5, 4, 3},
+	"SERIALNUMBER": {2, 5, 4, 5},
+	"L":            {2, 5, 4, 7},
+	"ST":           {2, 5, 4, 8},
+	"STREET":       {2, 5, 4, 9},
+	"POSTALCODE":   {2, 5, 4, 17},
+}
+
+func ParseSubjectStringToRdnSequence(subject string) (pkix.RDNSequence, error) {
+
+	dns, err := ldap.ParseDN(subject)
+	if err != nil {
+		return nil, err
+	}
+
+	// Traverse the parsed RDNSequence in REVERSE order as we use an LDAP parser which expects LDAP strings which use reverse notation.
```

Comment:
So actually  was mistaken when I wrote the comment. The parser, indeed `ldap.ParseDN`, itself doesn't reverse the order of the RDNs but that was done by the `.String()` function.

So, because the parser *does not* reverse the RDNs for us, we must do it as strings are supposed to be in the reverse order.

Reply:
Thanks for updating that comment

Yes, I remember now that the string representation is meant to be in reverse order as per the RFC 4514 https://datatracker.ietf.org/doc/html/rfc4514#section-2.1